core/src/main/java/org/mule/lifecycle/RegistryLifecycleManager.java

```diff
@@ -56,22 +46,20 @@ public RegistryLifecycleManager(String id, Registry object, MuleContext muleCont
 
     protected void registerPhases()
     {
-        RegistryLifecycleCallback callback = new RegistryLifecycleCallback();
+        final RegistryLifecycleCallback<Object> callback = new RegistryLifecycleCallback<>(this);
+        final LifecycleCallback<AbstractRegistryBroker> emptyCallback = new EmptyLifecycleCallback<>();
```

Comment:
some phases are applied by the muleContext directly. As stated above, this is not good but I choose to overlook it for now in order to decrease risk